src/request/change-password/ChangePassword.js

```diff
@@ -0,0 +1,211 @@
+        /** @type {HTMLFormElement} */
+        const $downloadFile = (document.getElementById(ChangePassword.Pages.DOWNLOAD_FILE));
+
+        // Elements
+        /** @type {HTMLFormElement} */
+        const $passwordGetter = (this.$enterPassword.querySelector('.password-box'));
+        /** @type {HTMLFormElement} */
+        const $passwordSetter = (this.$setPassword.querySelector('.password-setter-box'));
+        /** @type {HTMLDivElement} */
+        const $loginFileIcon = (this.$setPassword.querySelector('.login-file-icon'));
+        /** @type {HTMLAnchorElement} */
+        const $downloadLoginFile = ($downloadFile.querySelector('.download-login-file'));
+
+        // Components
+        this._passwordSetter = new PassphraseSetterBox($passwordSetter, { hideSkip: true });
+        this._loginFileIcon = new LoginFileIcon($loginFileIcon);
+        const downloadLoginFile = new DownloadLoginFile($downloadLoginFile);
+
+        this._passwordGetter = new PassphraseBox($passwordGetter, {
+            buttonI18nTag: 'passphrasebox-continue',
+            minLength: this._request.keyInfo.hasPin ? 6 : undefined,
+            hideCancel: true,
+            hideInput: !this._request.keyInfo.encrypted,
+        });
+
+        // Adapt to type of secret
+        if (this._request.keyInfo.type === Nimiq.Secret.Type.PRIVATE_KEY) {
+            this._loginFileIcon.setFileUnavailable(true);
+            /** @type {HTMLDivElement} */ (this.$enterPassword.querySelector('.nq-text')).classList.add('display-none');
+            /** @type {HTMLDivElement} */ (this.$setPassword.querySelector('.nq-text')).classList.add('hidden');
+        }
+
+        // Progress Indicators
+        // eslint-disable-next-line no-new
+        new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.ENTER_PASSWORD} .progress-indicator`),
+            4,
+            1,
+        );
+        this._progressIndicator = new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.SET_PASSWORD} .progress-indicator`),
+            4,
+            2,
+        );
+        // eslint-disable-next-line no-new
+        new ProgressIndicator(
+            document.querySelector(`#${ChangePassword.Pages.DOWNLOAD_FILE} .progress-indicator`),
+            4,
+            4,
+        );
+
+        // Events
+
+        this._passwordGetter.on(PassphraseBox.Events.SUBMIT, this._unlock.bind(this));
+
+        this._passwordSetter.on(PassphraseSetterBox.Events.ENTERED, () => {
+            let colorClass = '';
+            if (this._key.secret instanceof Nimiq.Entropy) {
+                const color = Iqons.getBackgroundColorIndex(
+                    new Nimiq.Address(
+                        // use color of first address as loginFile color
+                        this._key.deriveAddress(Constants.DEFAULT_DERIVATION_PATH).serialize(),
+                    ).toUserFriendlyAddress(),
```

Comment:
#229 adds `this._key.defaultAddress` to easily get the first address of a key. Reminder to replace this whole block with

```js
this._key.defaultAddress.toUserFriendlyAddress()
```